hal/inc/concurrent_hal.h

```diff
@@ -131,6 +131,25 @@ os_result_t os_thread_cleanup(os_thread_t thread);
  */
 os_result_t os_thread_yield(void);
 
+os_unique_id_t os_thread_unique_id(os_thread_t thread);
+
+os_thread_t os_thread_current();
+
+typedef struct {
+  uint8_t reserved;
```

Comment:
That's a question in the original PR as well :)
> It's currently only used within a single module and is not exported, so that shouldn't be a concern at the moment. If we later decide to export it, the first reserved becomes a version field as usual.